command/do_transition_test.go

```diff
@@ -1,60 +1,98 @@
-/*
-Copyright (C) 2018 Expedia Group.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
 package command
 
 import (
+	"encoding/json"
 	"github.com/ExpediaGroup/flyte-jira/client"
+	"github.com/HotelsDotCom/flyte-client/flyte"
+	"io/ioutil"
 	"net/http"
 	"reflect"
 	"testing"
 )
 
-func TestDoTransitionAsExpected(t *testing.T) {
-	client.SendRequest = func(request *http.Request, responseBody interface{}) (int, error) {
-		return http.StatusNoContent, nil
+type (
+	mockRequestBody struct {
+		TransitionId string `json:"id"`
 	}
 
-	var inputStruct = struct {
-		IssueId      string `json:"issueId"`
-		TransitionId string `json:"transitionId"`
-	}{"DEVEX-123", "881"}
-	input := toJson(inputStruct, t)
+	DoTransitionRequest struct {
+		Transition mockRequestBody `json:"transition"`
+	}
+)
```

Comment:
nit: is there any way to re use the `client.jira` structs instead of creating new ones?

Reply:
I guess yeah, will see how it goes. Thank you!!